x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/BlockRef.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import org.elasticsearch.core.Releasable;
+
+public class BlockRef implements Releasable {
```

Comment:
I was thinking you'd stick this in `Block.Ref` actually. It'll change fewer files at least. Not sure that matters though.